frontend/pkg/frontend/frontend.go

```diff
@@ -431,6 +509,227 @@ func (f *Frontend) ArmResourceCreateOrUpdate(writer http.ResponseWriter, request
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	versionedNodePool := versionedInterface.NewHCPOpenShiftClusterNodePool(nil)
+	if err = json.Unmarshal(body, versionedNodePool); err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteCloudError(writer, arm.NewUnmarshalCloudError(err))
+		return
+	}
+
+	if cloudError := versionedNodePool.ValidateStatic(); cloudError != nil {
+		f.logger.Error(cloudError.Error())
+		arm.WriteCloudError(writer, cloudError)
+		return
+	}
+	nodePoolName := request.PathValue(PathSegmentNodepoolName)
+	f.logger.Info(fmt.Sprintf("nodePoolName: %v", nodePoolName))
+
+	apiNodePool := buildInternalNodePool(versionedNodePool, nodePoolName)
+	newCsNodePool, err := f.BuildCSNodepool(ctx, apiNodePool)
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	req, err := f.CreateCSNodePool(clusterDoc.ClusterID, newCsNodePool)
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	nodePoolID := req.Body().ID()
+	nodePoolDoc.NodePoolID = nodePoolID
+	err = f.dbClient.SetNodePoolDoc(ctx, nodePoolDoc)
+	if err != nil {
+		f.logger.Error(fmt.Sprintf("failed to create nodepool document for resource %s: %v", nodePoolResourceID, err))
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	f.logger.Info(fmt.Sprintf("node pool document created for %s", nodePoolResourceID))
+
+	resp, err := json.Marshal(versionedNodePool)
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	_, err = writer.Write(resp)
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	writer.WriteHeader(http.StatusCreated)
+}
+
+func buildInternalNodePool(versionedNodePool api.VersionedHCPOpenShiftClusterNodePool, nodePoolName string) *api.HCPOpenShiftClusterNodePool {
+	apiNodePool := api.NewDefaultHCPOpenShiftClusterNodepool()
+	versionedNodePool.Normalize(apiNodePool)
+	apiNodePool.Name = nodePoolName
+
+	return apiNodePool
+}
+
+func (f *Frontend) DeleteNodePool(writer http.ResponseWriter, request *http.Request) {
+	ctx := request.Context()
+
+	versionedInterface, err := VersionFromContext(ctx)
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	f.logger.Info(fmt.Sprintf("%s: DeleteNodePool", versionedInterface))
+
+	// URL path is already lowercased by middleware.
+	nodePoolResourceID := request.URL.Path
+	clusterResourceID := clusterResourceIdFromNodePoolResourceID(nodePoolResourceID)
+
+	subscriptionID := request.PathValue(PathSegmentSubscriptionID)
+	clusterDoc, err := f.dbClient.GetClusterDoc(ctx, clusterResourceID, subscriptionID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			f.logger.Info(fmt.Sprintf("existing document not found for cluster %s when deleting node pool", clusterResourceID))
+			arm.WriteInternalServerError(writer)
```

Comment:
I think this should be a 404 error on delete.  Resource doesn't exist sort of thing, not internal server error.

Reply:
Sure. I forgot to modify the Copy/pasted block.Thanks.